lib/Component/VEvent.php

```diff
@@ -52,7 +52,10 @@ public function isInTimeRange(\DateTime $start, \DateTime $end) {
             //
             // See:
             // http://tools.ietf.org/html/rfc5545#page-54
-            $effectiveEnd = $this->DTEND->getDateTime();
+            $effectiveEnd = $this->DTEND->getDateTime($end->getTimezone());
+            if (!$this->DTEND->hasTime()) {
+                $effectiveEnd->modify('-1 second');
```

Comment:
Removing this and the other "-1" along with just doing a `<` comparison would at least also pass the existing tests, but not 100% sure if it might be wrong for any of the other cases.